factory/src/lib.rs

```diff
@@ -156,19 +193,24 @@ impl TokenFactory {
             .collect()
     }
 
-    pub fn get_whitelisted_tokens(&self, from_index: u64, limit: u64) -> Vec<TokenAccountId> {
+    pub fn get_whitelisted_token_tickers(&self, from_index: u64, limit: u64) -> Vec<TokenAccountId> {
         let token_ids = self.whitelisted_tokens.keys_as_vector();
         (from_index..std::cmp::min(from_index + limit, token_ids.len()))
             .filter_map(|token_id| token_ids.get(token_id))
             .collect()
     }
 
+    pub fn get_whitelisted_token(&self, token_id: TokenAccountId) -> (AccountId, FungibleTokenMetadata) {
+        let token = self.whitelisted_tokens.get(&token_id).expect("Token not found");
+        return (token.asset_id, token.metadata);
+    }
```

Comment:
Need to add `Serialize` for `WhitelistedToken`
```suggestion
    pub fn get_whitelisted_token(&self, token_id: TokenAccountId) -> WhitelistedToken {
        self.whitelisted_tokens.get(&token_id).expect("Token not found")
    }
```